sdk/core/azure-core/src/main/java/com/azure/core/implementation/ImplUtils.java

```diff
@@ -94,6 +99,47 @@ private static long tryParseLong(String value) {
         }
     }
 
+    /**
+     * Writes a {@link ByteBuffer} into an {@link OutputStream}.
+     * <p>
+     * This method provides writing optimization based on the type of {@link ByteBuffer} and {@link OutputStream}
+     * passed. For example, if the {@link ByteBuffer} has a backing {@code byte[]} this method will access that directly
+     * to write to the {@code stream} instead of buffering the contents of the {@link ByteBuffer} into a temporary
+     * buffer.
+     *
+     * @param buffer The {@link ByteBuffer} to write into the {@code stream}.
+     * @param stream The {@link OutputStream} where the {@code buffer} will be written.
+     * @throws IOException If an I/O occurs while writing the {@code buffer} into the {@code stream}.
+     */
+    public static void writeByteBufferToStream(ByteBuffer buffer, OutputStream stream) throws IOException {
```

Comment:
nit: is there a better place for this than `ImplUtils`?

Reply:
I think so, but given this is in implementation we can move it later, such as if we do an implementation refactoring of data related methods into DataUtils